### PR TITLE
Fix report arguments handling in CSV export.

### LINKED ISSUE #5503: Categories Emailed Report contains items from order categories
**Describe the bug**
The emailed report from Analytics > Categories does not match what's shown on the screen.

**To Reproduce**
1. Go to Analytics > Categories
2. Choose a single category to filter
3. If necessary, edit the rows per page to span several pages to trigger an email rather than a direct download
4. Notice that the emailed report includes additional products not in the selected category.

**Expected behavior**
I expect the emailed report to match what's on my screen.

What seems to be happening is that any products purchased together with products in that category are added to the emailed report even though they don't show up in the list on the screen.

**Screenshots**

**Desktop (please complete the following information):**

* OS: Mac OS Catalina
* Browser Chrome
* Version 83

**Additional context**
Reported in [the forums](https://wordpress.org/support/topic/analytics-export-direct-download-vs-email/).

**Replicated**
Screenshot: https://d.pr/i/jVS5Jv
CSV file: https://d.pr/f/cqt1rf

### PR DESCRIPTION
**Pull request by jeffstieler**
Fixes #5503.

This PR fixes the CSV Report Export handling of report arguments/parameters. Before this change, the arguments were not mapped (in Controller) and were not sanitized (using the schema defined in the `Controller::get_collection_params`).

In writing the test case, I found a bug with the handling of empty category results for products (the `-1` placeholder ID used to facilitate the query was showing up in results) - this PR includes a fix for that as well.

### Detailed test instructions:

- Go to Analytics > Categories
- Choose a single category to filter
- If necessary, edit the rows per page to span several pages to trigger an email rather than a direct download
- Notice that the emailed report **does not include** additional products not in the selected category.

### CODE REVIEW

File: tests/reports/class-wc-tests-reports-products.php [hunk 1]
```diff
@@ -414,4 +415,75 @@ public function test_populate_and_refund() {
...
 		$this->assertEquals( $expected_data, $query->get_data() );
 	}
+
+	/**
+	 * Test that filters get properly parsed for CSV exports.
+	 * See: https://github.com/woocommerce/woocommerce-admin/issues/5503.
+	 *
+	 * @since 3.5.0
+	 */
+	public function test_report_export_arguments() {
```
Comment (joshuatf) [inline tests/reports/class-wc-tests-reports-products.php:425]: Thanks for these tests!

File: src/API/Reports/Products/DataStore.php [hunk 2]
```diff
@@ -303,8 +303,9 @@ public function get_data( $query_args ) {
 			$this->add_sql_query_params( $query_args );
 
 			if ( count( $included_products ) > 0 ) {
-				$total_results = count( $included_products );
-				$total_pages   = (int) ceil( $total_results / $params['per_page'] );
+				$filtered_products = array_diff( $included_products, array( '-1' ) );
```
Comment (joshuatf) [inline src/API/Reports/Products/DataStore.php:306]: 👍

**Review by joshuatf — APPROVED** [review]
Great fix, Jeff!  This is testing well on my end.

It would be great to have a test around the erroneously included products bug itself, but not a blocker for getting this merged as I know you have other things on your plate at the moment.

**Comment by jeffstieler** [comment]
> It would be great to have a test around the erroneously included products bug itself

The included test case covers this by filtering for the unused product category and verifying there are no results.

**Pull request merged**